src/screens/register/index.js

```diff
@@ -1,182 +1,64 @@
-    const classes = {
-      root: ''
-    }
-    const steps = this.getSteps()
-    const {activeStep} = this.state
-    const currentStep = this.getCurrentStep()
-
-    return (
-      <Formik
-        initialValues={{email: '', password: ''}}
-        validate={values => {
-          const errors = {}
-          if (!values.email) {
-            errors.email = 'Required'
-          } else if (
-            !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
-          ) {
-            errors.email = 'Invalid email address'
-          }
-          return errors
-        }}
-        onSubmit={(values, {setSubmitting}) => {
-          setTimeout(() => {
-            /* Alert(JSON.stringify(values, null, 2)) */
-            setSubmitting(false)
-          }, 400)
-        }}
-      >
-        {(
-          {
-            /* Values, */
-            /* errors, */
-            /* touched, */
-            /* handleChange, */
-            /* handleBlur, */
-            handleSubmit,
-            isSubmitting
-            /* And other goodies */
-          }
-        ) => (
-          <form onSubmit={handleSubmit}>
-            <div className={classes.root}>
-              <Stepper activeStep={activeStep} alternativeLabel>
-                {currentStep.map(label => {
-                  return (
-                    <Step key={label}>
-                      <StepLabel>{label}</StepLabel>
-                    </Step>
-                  )
-                })}
-              </Stepper>
-              <button type="submit" disabled={isSubmitting}>
-                Submit
-              </button>
-              <div>
-                {this.state.activeStep === steps.length ? (
-                  <div>
-                    <Typography className={classes.instructions}>
-                      All steps completed
-                    </Typography>
-                    <Button onClick={this.handleReset}>Reset</Button>
-                  </div>
-                ) : (
-                  <div>
-                    <Typography className={classes.instructions}>
-                      {getStepContent(activeStep)}
-                    </Typography>
-                    <div>
-                      <Button
-                        disabled={activeStep === 0}
-                        onClick={this.handleBack}
-                        className={classes.backButton}
-                      >
-                        Back
-                      </Button>
-                      <Button
-                        variant="contained"
-                        color="primary"
-                        onClick={this.handleNext}
-                      >
-                        {activeStep === steps.length - 1 ? 'Finish' : 'Next'}
-                      </Button>
-                    </div>
-                  </div>
-                )}
-              </div>
-            </div>
-          </form>
-        )}
-      </Formik>
-    )
-  }
-}
+      return <Steps {...props} />
```

Comment:
i didn't quite get how can the steps be rendered individually with only this line. wont they end up being rendered all at once in the same screen?

Reply:
@gugutz The steps component contain the logic that decides which component will be rendered. 

```js
    const CurrentStep = steps[this.state.currentStep]

    return (
      <div>
        <CurrentStep {...this.props} />
```
It gets the current step and select it from the list of steps.